File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/config/serializer/TextSerializer.java
package de.jalumu.magma.platform.base.config.serializer;

import de.exlll.configlib.Serializer;
import de.jalumu.magma.text.Text;

import java.util.HashMap;

public class TextSerializer implements Serializer<Text, HashMap<String, Object>> {
    @Override
    public HashMap<String, Object> serialize(Text element) {
        HashMap<String, Object> map = new HashMap<>();
        map.put("settings.parseColorCodes", element.parseColorCodes());
        map.put("settings.parsePlaceholders", element.parsePlaceholders());
        map.put("settings.allowMiddleware", element.allowMiddleware());
        map.put("text.default", element.getDefaultText());

        element.getTranslations().forEach((locale, s) -> {
            map.put("text.lang." + locale.getISO3Country(), s);
        });

        map.put("text.default", element.getDefaultText());

        return map;
    }

    @Override
    public Text deserialize(HashMap<String, Object> element) {
        return null;
    }
}

File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/config/serializer/TextSerializer.kt
package de.jalumu.magma.platform.base.config.serializer

import de.exlll.configlib.Serializer
import de.jalumu.magma.text.Text
import java.util.*

class TextSerializer : Serializer<Text?, HashMap<String, Any>> {
    override fun serialize(element: Text?): HashMap<String, Any> {
        val map = HashMap<String, Any>()
        map["settings.parseColorCodes"] = element!!.parseColorCodes()
        map["settings.parsePlaceholders"] = element.parsePlaceholders()
        map["settings.allowMiddleware"] = element.allowMiddleware()
        map["text.default"] = element.defaultText
        element.translations.forEach { (locale: Locale, s: String) -> map["text.lang." + locale.isO3Country] = s }
        map["text.default"] = element.defaultText
        return map
    }

    override fun deserialize(element: HashMap<String, Any>): Text? {
        return null
    }
}
